### PR TITLE
Use public cloud for digital ocean and secure with TLS

### PR DESCRIPTION
**Pull request by sheerun**
"Private network" on digital ocean is really a shared private network.

It means all other hosts in datacenter, even of other users, can access deployed nodes on digitalocean. It means deploying on public ips is not less secure than deploying on private ones.

In this setup we additionaly secure etcd with wildcard tls certs.

As an additional bonus, we can not deploy servers across datacenters.

Also, we generate initial ssh key for digital ocean dynamically, for better security, and export it to "id_rsa" file for easy ssh access.

### CODE REVIEW

**Comment by sheerun** [comment]
This closes #651

File: terraform/digitalocean/main.tf [hunk 1]
```diff
@@ -13,11 +12,82 @@ provider "digitalocean" {
...
+  algorithm = "RSA"
+}
+
+resource "tls_self_signed_cert" "ca" {
+  key_algorithm = "RSA"
+  private_key_pem = "${tls_private_key.ca.private_key_pem}"
+
+  subject {
+    common_name = "*"
+    organization = "skybase"
```
Comment (tayzlor) [inline terraform/digitalocean/main.tf]: minor: wonder if we should make this  ```${var.org_name}``` and default it to "apollo" or similar.

**Comment by sheerun** [comment]
Made the fix you suggested :)

**Comment by tayzlor** [comment]
Thanks!

**Pull request merged**